### PR TITLE
fix(Core/Conditions): Implement CONDITION_DAILY_QUEST_DONE

### PR DESCRIPTION
**Pull request by Kitzunu**
## CHANGES PROPOSED:
* Implement Player::IsDailyQuestDone

* Implement CONDITION_DAILY_QUEST_DONE

* Imported from TC

## ISSUES ADDRESSED:
- Ref https://github.com/azerothcore/azerothcore-wotlk/issues/1372

## TESTS PERFORMED:
- Builds

## HOW TO TEST THE CHANGES:

```sql
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=19 AND `SourceGroup`=0 AND `SourceEntry`=40;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES 
(19, 0, 40, 0, 0, 43, 0, 11153, 0, 0, 0, 0, 0, '', '');
```` 
THIS IS A DUMMY SQL. MAKE SURE TO DELETE IT AFTER TESTING.

1. Apply SQL 
2. .go c id 241
and see that he will offer only one quest (out of 2)
3. .q add 11153 (This is a daily quest)
4. .q comp 11153
5. .q rew 11153
6. See that he now offers two quests

Tada PR Works :) 

## KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -23566,6 +23566,24 @@ void Player::SetDailyQuestStatus(uint32 quest_id)
...
+            {
+                found = true;
+                break;
+            }
+        }
+    }
+
+    return found;
+}
+
```
Comment (Helias) [inline src/server/game/Entities/Player/Player.cpp:23586]: you could avoid `break` and put `return true`, and at the end `return false`, in this way you can remove the variable `found`.
I mean, in this way:
```C++
bool Player::IsDailyQuestDone(uint32 quest_id)
{
    if (sObjectMgr->GetQuestTemplate(quest_id)) {
        for (uint32 quest_daily_idx = 0; quest_daily_idx < PLAYER_MAX_DAILY_QUESTS; ++quest_daily_idx) {
            if (GetUInt32Value(PLAYER_FIELD_DAILY_QUESTS_1 + quest_daily_idx) == quest_id) {
              return true;
            }
        }
    }

    return false;
}
```

**Comment by Kitzunu** [comment]
Woop got this to work too :)

File: src/server/game/Conditions/ConditionMgr.cpp [hunk 2]
```diff
@@ -361,6 +361,12 @@ bool Condition::Meets(ConditionSourceInfo& sourceInfo)
                     condMeets = unit->IsInWater();
                 break;
             }
+        case CONDITION_DAILY_QUEST_DONE:
+            {
+                if (Player* player = object->ToPlayer())
+                    condMeets = player->IsDailyQuestDone(ConditionValue1);
```
Comment (FrancescoBorzi) [inline src/server/game/Conditions/ConditionMgr.cpp:367]: brackets please

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by Kitzunu** [comment]
Everything works as it should, good suggestion @Helias

**Comment by FrancescoBorzi** [comment]
looks good to me, it has been tested by the author and shouldn't affect any existing features

**Pull request merged**